hedera-node/hedera-app/src/test/java/com/hedera/node/app/utils/HashTreeManagerTest.java

```diff
@@ -0,0 +1,196 @@
+    private final String SHA_384_ROOT_HASH =
+            "360526b48cb86c82e7af61c6a880afcfce4c738d02c4109338a44fe2994f1a8f3f9ee832693ad0f723be3b2777b97fcb";
+
+    private final Map<BlockHeader, MerkleNode> inputRootsByBlock = new HashMap<>();
+    private final Map<BlockHeader, MerkleNode> outputRootsByBlock = new HashMap<>();
+    private final Set<Pair<MerkleNode, Bytes>> completedSubtreeHashes = new HashSet<>();
+
+    @Test
+    public void testHashTreeManager() {
+        // Example elements to be processed
+        List<String> elements = Arrays.asList("Event1", "Transaction1", "Transaction2", "Event2", "Transaction3");
+
+        // Process the elements
+        hashTreeManager.processNodes(elements);
+
+        // Calculate and print the Merkle root hash
+        Bytes merkleRootHash = hashTreeManager.calculateMerkleRootHash();
+        System.out.println("Merkle Root Hash: " + merkleRootHash.toString());
+        assertThat(merkleRootHash).isNotNull();
+        assertThat(merkleRootHash.toString())
+                .isEqualTo(
+                        "49d156f8dda7219898d98d48a5bdc8df453c993890c7f1dbe2628361ff6fbb19553ad6258d0431691d20bd8b2cb0ab07");
+    }
+
+    @Test
+    void testCalculateMerkleRoot() throws Exception {
+        List<String> elements = Arrays.asList("0", "1", "2", "3", "4", "5", "6", "7");
+        String rootHash = hashTreeManager.calculateMerkleRootOnLeafNodes(elements);
+        assertThat(rootHash).isNotNull();
+        assertThat(rootHash).isEqualTo(SHA_384_ROOT_HASH);
+    }
+
+    @Test
+    void testCollisionResistance() {
+        boolean foundCollision = false;
+        // A collision occurs when two different inputs produce the same hash output.
+        for (long i = 0L; i < 1000L; i++) {
+            for (long j = 1000L; j < 2000L; j++) {
+                String hash1 = hashTreeManager.sha384("input" + i).toString();
+                String hash2 = hashTreeManager.sha384("input" + j).toString();
+                if (hash1.equals(hash2)) {
+                    foundCollision = true;
+                    break;
+                }
+            }
+            if (foundCollision) break;
+        }
+        assertThat(foundCollision).isFalse();
+    }
+
+    @Test
+    void validateHashingWorksAsExpected() {
+        boolean foundCollision = false;
+        // Ensure the same inputs to the hash algorithm produce the same hash output.
+        for (long i = 0L; i < 10000L; i++) {
+            for (long j = 0L; j < 10000L; j++) {
+                String hash1 = hashTreeManager.sha384("input" + i).toString();
+                String hash2 = hashTreeManager.sha384("input" + j).toString();
+                if (hash1.equals(hash2)) {
+                    foundCollision = true;
+                    break;
+                }
+            }
+            if (foundCollision) break;
+        }
+        assertThat(foundCollision).isTrue().withFailMessage("A collision was found.");
+    }
```

Comment:
This feels like testing the sha384 algorithm, which seems unnecessary. These two tests also require the `sha394()` method to be part of `HashTreeManager`'s public API, which also seems unnecessary. What benefits do these tests provide? We're already assuming sha384 won't collide, otherwise we'd use a different algorithm. Maybe I'm missing something?